test/e2e/util_test.go

```diff
@@ -520,6 +520,31 @@ func probe(timeout, period, success, failure int) *corev1.Probe {
 	}
 }
 
+// updateRouteWithRetryOnConflict gets a fresh copy of the named route,
+// calls mutateRouteFn() where callers can modify fields of the route,
+// and then updates the route object. If there is a conflict error
+// on update then the complete sequence of get, mutate, and update
+// is retried until timeout is reached.
+func updateRouteWithRetryOnConflict(t *testing.T, name types.NamespacedName, timeout time.Duration, mutateRouteFn func(route *routev1.Route)) error {
+	t.Helper()
+	route := routev1.Route{}
+	return wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
```

Comment:
I know this function is used everywhere but [it's deprecated](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#PollImmediate). We better start using `PollUntilContextTimeout` ([example](https://github.com/openshift/cluster-ingress-operator/blob/76cfd8a48e4dc98dfd5787d4facaacc32616522d/test/e2e/util_test.go#L981)).